lib/esm/embed.rb

```diff
@@ -19,6 +19,79 @@ def self.build(type = nil, **, &)
       ESM::Embed.new(type, **, &)
     end
 
+    # This is defensively written because this can receive user provided data
+    def self.from_hash(hash, &block)
+      raise TypeError, "Expected Hash, got #{hash.class}" unless hash.is_a?(Hash)
+
+      hash = hash.with_indifferent_access
+      new do |embed|
+        ###########
+        # Author
+        # Supports string or HashMap with full options
+        if (author = hash[:author]) && author.present?
+          author = ESM::Arma::HashMap.from(author).presence || {name: author}
+          author = author.slice(:name, :url, :icon_url).symbolize_keys
+          embed.set_author(**author)
+        end
+
+        ###########
+        # Title
+        if (title = hash[:title]) && title.present?
+          embed.title = title.to_s
```

Comment:
Why does author get a setter but the remainder of attributes are directly manipulated?

Reply:
Remember, Ruby setters are not methods that start with `set`  lol `def title=(text)` :wink: 

This class is very old but IIRC, the naming is this way because the title itself is a string but the author is a separate class. Plus, I can't use keyword arguments (kwargs) with setters as that they can only accept a single argument